File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-5.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int l = 0;
        int r = 0;
        int res = k;
        map<char,int>mp;
        multiset<int>multi;
        while (r<s.size()){
            mp[s[r]]++;
            multi.insert(mp[s[r]]);
            if (r-l+1 - *multi.rbegin() > k){
                multi.erase(mp[s[l]]);
                mp[s[l]]--;
                multi.insert(mp[s[l]]);
                l++;
            }

            res = max(res,r-l+1);
            r++;

        }
        return res;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-5.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        seen = set()
        res = 0
        i = 0
        for r in range(len(s)):
            if s[r] in seen:
                while s[i]!= s[r]:
                    seen.remove(s[i])
                    i+=1
                else:
                    i+=1
            else:
                seen.add(s[r])
            res = max(res,r-i+1)
        
        return res